### PR TITLE
Flatten pick and omit APIs in CallWithChatComposite

### PR DESCRIPTION
**Pull request by JamesBurnside**
# What
Flatten following CallWithChatComposite APIs that were using Pick and Omit:
* CallWithChatAdapterManagement
* CallWithChatAdapterUiState
* CallWithChatClientState
* CallWithChatControlOptions

# Why
Azure Review Board guidance -- doing so improves intellisense, improves doc generation and improves dev experience when looking at the API in the .d.ts file.

# How Tested
n/a

### CODE REVIEW

File: packages/react-composites/review/react-composites.api.md [hunk 1]
```diff
@@ -539,8 +569,15 @@ export interface CallWithChatCompositeStrings {
 
 // @beta
 export interface CallWithChatControlOptions extends Pick<CallControlOptions, 'cameraButton' | 'microphoneButton' | 'screenShareButton' | 'displayType'> {
```
Comment (prprabhu-ms) [inline packages/react-composites/review/react-composites.api.md:571]: I'm confused. So we now have *both* `extends Pick<CallControlOptions, 'cameraButton'...` and an explicit `cameraButton?: boolean` field? Worse, the two can diverge without build failure.

What I was expecting here:
* Drop the `Pick<...>` entirely.
* Add internal build-time check that `CallWithChatControlOptions` indeed extends `Pick <..>` etc.

Comment (JamesBurnside) [reply packages/react-composites/review/react-composites.api.md:571]: oh nope, that was not removed by accident! but I really like that internally we still use Pick, I will update with that

Comment (prprabhu-ms) [reply packages/react-composites/review/react-composites.api.md:571]: See https://github.com/Azure/communication-ui-library/pull/1560/files#diff-db700faf40319449d13fa41bfdf20a7dc853a55c7f6a8e6f8dd3bb09d85a51d5 for an example.

**Comment by JamesBurnside** [comment]
Closing as need to make an update to this

**Pull request closed without merging**